src/auth/lib/index.ts

```diff
@@ -56,28 +56,38 @@ export async function submitRegistration(
   type: "search" | "browser",
 ) {
   const path = type === "search" ? "/register/search" : "/register";
+  const body = {
+    advertiser: {
+      billingEmail: form.user.email,
+      name: form.advertiser.name,
+      url: form.advertiser.url,
+      description: form.advertiser.description,
+      marketingChannel:
+        form.advertiser.marketingChannel === "other"
+          ? `other: ${form.advertiser.other}`
+          : form.advertiser.marketingChannel,
+      vertical: form.advertiser.vertical,
+    },
+    user: form.user,
+    country: form.country,
+    mediaSpend: form.mediaSpend,
+  };
+
+  if (type === "browser") {
+    (body.advertiser as any)["vertical"] =
+      form.advertiser.vertical === "Other"
+        ? undefined
+        : form.advertiser.vertical;
+  }
+
```

Comment:
```suggestion
  const body = {
    advertiser: {
      billingEmail: form.user.email,
      name: form.advertiser.name,
      url: form.advertiser.url,
      description: form.advertiser.description,
      marketingChannel:
        form.advertiser.marketingChannel === "other"
          ? `other: ${form.advertiser.other}`
          : form.advertiser.marketingChannel,
      vertical:
        form.advertiser.vertical === "Other"
          ? undefined
          : form.advertiser.vertical,
    },
    user: form.user,
    country: form.country,
    mediaSpend: form.mediaSpend,
  };

```

avoid the need for the "as any" cast. code compiles but has not been tested!